### ISSUE TITLE
Illegal instructions

### ISSUE THREAD

**Issue by weedmank** [alaasal/GPCore#5]
I noticed the following is unused code.  If the lower 2 bits of any opcode != 2'b11 then the instruction is a Compressed Instruction and should cause a trap if  you don't support Compressed Instructions or instructions larger than 32 bits

assign noOp = ~(|op); // if op code = 000000 then set opcode0 to 1

**Comment by alaasal** [comment]
This should be closed as it is duplicate to #4 ?

**Comment by weedmank** [comment]
Since you do not decode Atomic Instructions, do you cause a trap since they would be illegal instructions. There may be other RV32i instructions from unimplemented extensions you don't decode that would have the same issue.

**Comment by alaasal** [comment]
Yes I totally get what you are saying but illegal intrs trap is part of the privileged isa support that is not yet merged to master.

**Comment by moustafamd** [comment]
> Since you do not decode Atomic Instructions, do you cause a trap since they would be illegal instructions. There may be other RV32i instructions from unimplemented extensions you don't decode that would have the same issue.

We merged the privileged ISA implementation to master and the illegal instruction condition updated so any unimplemented instruction in our core will raise an illegal instruction exception and cause a trap to be handled by the trap handler.
